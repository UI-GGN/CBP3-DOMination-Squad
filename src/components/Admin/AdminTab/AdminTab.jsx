import hamburger from '../../../assets/menu.png';
import { Container, HamburgerIcon, Title } from './AdminTab.style';

const AdminTab = ({ selected, onIconPress, isHeaderVisible }) => {
  return (
    <Container data-testid="container">
      {!isHeaderVisible && <HamburgerIcon data-testid="hamburger-icon" src={hamburger} onClick={() => onIconPress()} />}
      <Title data-testid="admin-tab-title">{selected}</Title>
    </Container>
  );
};

export default AdminTab;
