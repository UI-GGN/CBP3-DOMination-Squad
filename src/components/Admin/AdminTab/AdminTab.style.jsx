import { primary, dark } from '../../colors.json';
import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: row;
  background-color: ${primary};
  align-items: center;
  height: 8vh;
`;

export const HamburgerIcon = styled.img`
  height: 60%;
  width: 45px;
  margin-left: 8px;
  cursor: pointer;
`;

export const Title = styled.h1`
  font-size: 38px;
  margin: 16px 24px;
  color: ${dark};
`;
