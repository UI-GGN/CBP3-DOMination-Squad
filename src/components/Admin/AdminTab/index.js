import AdminTab from './AdminTab';

export default AdminTab;
