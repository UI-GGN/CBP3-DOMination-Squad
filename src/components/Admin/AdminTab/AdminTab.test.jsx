import AdminTab from './AdminTab';
import { render, fireEvent } from '@testing-library/react';
import { describe, it, vi, expect } from 'vitest';

describe('Admin Tab', () => {
  it('should render Admin Tab component', () => {
    const { container, getByTestId, queryByTestId } = render(<AdminTab selected="Requests" isHeaderVisible={true} />);

    expect(getByTestId('container')).toBeInTheDocument();
    expect(queryByTestId('hamburger-icon')).not.toBeInTheDocument();
    expect(getByTestId('admin-tab-title')).toBeInTheDocument();
    expect(getByTestId('admin-tab-title')).toHaveTextContent('Requests');
    expect(container).toMatchSnapshot();
  });

  it('should render Admin Tab component with hamburger icon when isHeaderVisible is true', () => {
    const { queryByTestId } = render(<AdminTab selected="Requests" isHeaderVisible={false} />);

    expect(queryByTestId('hamburger-icon')).toBeInTheDocument();
  });

  it('should call onIconPress when the hamburger icon is clicked', () => {
    const mockOnIconPress = vi.fn();
    const { getByTestId } = render(
      <AdminTab selected="Requests" onIconPress={mockOnIconPress} isHeaderVisible={false} />
    );

    fireEvent.click(getByTestId('hamburger-icon'));

    expect(mockOnIconPress).toHaveBeenCalledTimes(1);
  });
});
