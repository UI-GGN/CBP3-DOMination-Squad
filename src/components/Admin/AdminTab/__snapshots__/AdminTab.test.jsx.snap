// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Admin Tab > should render Admin Tab component 1`] = `
<div>
  <div
    class="sc-beqWaB jGrQuz"
    data-testid="container"
  >
    <h1
      class="sc-dmqHEX dsssyt"
      data-testid="admin-tab-title"
    >
      Requests
    </h1>
  </div>
</div>
`;
